### DISCUSSION TITLE
Could we make a new note type such as ChatNote?

### DISCUSSION THREAD

**Discussion by WantToLearnJapanese** [TriliumNext/Trilium#3857] [Ideas]
I want to integrate ChatGPT by using their API.

The reason behind my purpose is it is such a pain to structure conversations between ChatGPT. 
Using Trilium Notes, I want to organize my conversations.

In the current state of Trilium implementation, is it possible to make such a note type?

**Comment by zadam** [comment]
I don't know the API, but I imagine it should be possible to integrate it in some way. Using e.g. plain code notes to store the conversation, perhaps a custom widget to render the response, just "send" button. There are many options, I don't see why.

Having said that, I have no plans to provide a built-in support. Sounds cool, though.

**Reply by WantToLearnJapanese** [reply]
Oh surely not, I didn't mean built-in support. I was curious if it is possible with exposed APIs. I mean the new type of note.

**Reply by WantToLearnJapanese** [reply]
@zadam Could you answer me? Sorry to bother you.

**Reply by zadam** [reply]
Scripts can't define new note types, but a chat can be stored in a simple code note. You can make then custom widgets and such, call APIs, should be feasible.

**Comment by SiriusXT** [comment]
As far as I know, obsidian supports chatgpt because of plugins developed by third-party authors. A plug-in marketplace facilitates the creation of such plug-ins.

**Reply by WantToLearnJapanese** [reply]
I know Obsidian has such a plugin. What is your point? We are talking about Trilium.

**Reply by SiriusXT** [reply]
I hope trilium can also have its own plug-in store, which can promote more enthusiasts to join

**Reply by WantToLearnJapanese** [reply]
> I hope trilium can also have its own plug-in store, which can promote more enthusiasts to join

I agree that it will be nice to have a plugin market but not sure if it will boost more developers to join?

**Reply by soulsands** [reply]
@WantToLearnJapanese  Hi, here is a chatgpt plugin you might be interested in. https://github.com/soulsands/trilium-chat
